### PR TITLE
adding in source/target version

### PR DESCRIPTION
**Pull request by walterddr**
intellij imported profile didn't set source/target version properly. explicitly define it in pluginManagement section.

### CODE REVIEW

**Review by xiangfu0 — APPROVED** [review]

**Comment by kkrugler** [comment]
So this would target Java 11 for compilation, yes? Is this a hard break with Java 8 compatibility?

File: pom.xml [hunk 1]
```diff
@@ -282,6 +282,8 @@
               <version>3.8.0</version>
               <configuration>
                 <release>${jdk.version}</release>
+                <source>${jdk.version}</source>
+                <target>${jdk.version}</target>
```
Comment (richardstartin) [inline pom.xml:286]: `-release N` means `-source N -target N -bootclasspath N` - so this should be unnecessary unless you are using JDK8?

Comment (walterddr) [reply pom.xml:286]: i thought so too but for some reason using adoptopenjdk-11 with intellij it defaults source version to 7

Comment (richardstartin) [reply pom.xml:286]: Could be an IntelliJ bug (making this change the pragmatic thing to do)

**Comment by walterddr** [comment]
> So this would target Java 11 for compilation, yes? Is this a hard break with Java 8 compatibility?

nope. there's a specific profile that enables jdk8 support

**Comment by richardstartin** [comment]
Note that Java 8 compatibility does not require building with JDK8, it requires producing classfiles with version 52.0 linked against the JDK8 bootclasspath, so that the classes can be loaded on JDK8. This can be done with later versions of javac by setting the `release` flag. The advantage of using later versions of javac is that you can compile code to be packaged in Multi-Release jars, so you could, say, use foreign memory access on JDK17 but use `sun.misc.Unsafe` on JDK8. The requirement to _build with JDK8_ precludes this, or at the very least makes it much harder than it needs to be. I will create an issue to track this.

File: pom.xml (hunk 1 above)
Comment (xiangfu0) [reply pom.xml:286]: I think it's the issue of intellij. intellij doesn't translate this, so it will mess up the IDE.

Comment (walterddr) [reply pom.xml:286]: yeah but for now i dont think there's any better solutions than what i am doing here. should we just merge this one first?

**Comment by walterddr** [comment]
> Note that Java 8 compatibility does not require building with JDK8, it requires producing classfiles with version 52.0 linked against the JDK8 bootclasspath, so that the classes can be loaded on JDK8. This can be done with later versions of javac by setting the `release` flag. The advantage of using later versions of javac is that you can compile code to be packaged in Multi-Release jars, so you could, say, use foreign memory access on JDK17 but use `sun.misc.Unsafe` on JDK8. The requirement to _build with JDK8_ precludes this, or at the very least makes it much harder than it needs to be. I will create an issue to track this.

totally agree. should we just do `<release>8</release>` on higher jdk versions? ( i felt like we should, since currently we only test them in CI).

**Pull request merged**